go-controller/pkg/ovnwebhook/nodeadmission.go

```diff
@@ -0,0 +1,169 @@
+package ovnwebhook
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"strings"
+
+	hotypes "github.com/ovn-org/ovn-kubernetes/go-controller/hybrid-overlay/pkg/types"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/types"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"
+	"golang.org/x/exp/maps"
+	corev1 "k8s.io/api/core/v1"
+	apiequality "k8s.io/apimachinery/pkg/api/equality"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/sets"
+	nodeutil "k8s.io/component-helpers/node/util"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+type checkNodeAnnot func(v annotationChange, nodeName string) error
+
+// commonNodeAnnotationChecks holds annotations allowed for ovnkube-node:<nodeName> users in non-IC and IC environments
+var commonNodeAnnotationChecks = map[string]checkNodeAnnot{
+	util.OvnNodeHostAddresses:            nil,
+	util.OvnNodeL3GatewayConfig:          nil,
+	util.OvnNodeManagementPortMacAddress: nil,
+	util.OvnNodeIfAddr:                   nil,
+	util.OvnNodeChassisID: func(v annotationChange, nodeName string) error {
+		if v.action == removed {
+			return fmt.Errorf("%s cannot be removed", util.OvnNodeChassisID)
+		}
+		if v.action == changed {
+			return fmt.Errorf("%s cannot be changed once set", util.OvnNodeChassisID)
+		}
+		return nil
+	},
+	util.OvnNodeZoneName: func(v annotationChange, nodeName string) error {
+		// it is allowed for the annotation to be set to "global" or <nodeName> initially
+		if v.action == added && (v.value == types.OvnDefaultZone || v.value == nodeName) {
+			return nil
+		}
+
+		// it is allowed for the annotation to be changed to <nodeName>
+		if v.action == changed && v.value == nodeName {
```

Comment:
but what if we have our backdoor that lets us move the IC deployment back to global mode (1 zone). @ricky-rav FYI

Or what if someone manually downgrades to 4.13 and the webhook is present there? I guess an admin would need to clear the value of the annotation before trying to upgrade again to 4.14?

Reply:
I have the same feeling, restricting the possible flows mean that we restrict the corner cases in which those might be required. This check was specifically added so the zone cannot be changed to something that could compromise other nodes. I am open to any ideas on how we could improve this check.

Reply:
so then we should just allow global or the zones own name, and not enforce only from global -> node name?